pkg/operator/ceph/version/version.go

```diff
@@ -0,0 +1,118 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type CephVersion struct {
+	Major int
+	Minor int
+	Extra int
+}
+
+const (
+	unknownVersionString = "<unknown version>"
+)
+
+var (
+	Luminous = CephVersion{12, 0, 0}
+	Mimic    = CephVersion{13, 0, 0}
+	Nautilus = CephVersion{14, 0, 0}
+
+	// supportedVersions are production-ready versions that rook supports
+	supportedVersions   = []CephVersion{Luminous, Mimic}
+	unsupportedVersions = []CephVersion{Nautilus}
+	// allVersions includes all supportedVersions as well as unreleased versions that are being tested with rook
+	allVersions = append(supportedVersions, unsupportedVersions...)
+
+	// for parsing the output of `ceph --version`
+	versionPattern = regexp.MustCompile(`ceph version (\d+)\.(\d+)\.(\d+)`)
+)
+
+func (v *CephVersion) String() string {
+	return fmt.Sprintf("%d.%d.%d %s",
+		v.Major, v.Minor, v.Extra, v.ReleaseName())
+}
+
+func (v *CephVersion) ReleaseName() string {
+	switch v.Major {
+	case Nautilus.Major:
+		return "nautilus"
+	case Mimic.Major:
+		return "mimic"
+	case Luminous.Major:
+		return "luminous"
+	default:
+		return unknownVersionString
+	}
+}
+
+func ExtractCephVersion(src string) (*CephVersion, error) {
+	m := versionPattern.FindStringSubmatch(src)
+	if m == nil {
```

Comment:
I don't see a consensus on whether to support master builds without version info. I think we should support this case so devs working on Ceph code changes can iterate faster with Rook.

```[nwatkins@daq build]$ bin/ceph --version
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
ceph version Development (no_version) nautilus (rc)
```

Reply:
Actually, I changed my mind about not approving based on this. Even if we decide to support Ceph builds without versions, that can be a separate PR.